api/handler.go

```diff
@@ -301,11 +301,21 @@ func (h *apiHandler) queryUsage() http.HandlerFunc {
 		}
 
 		userId, ok := r.Context().Value(userIdContextKey).(string)
+		isCallerAdmin, _ := r.Context().Value(isCallerAdminContextKey).(string)
+
 		if !ok {
 			respondError(rw, http.StatusInternalServerError, errors.New("request not authenticated"))
 			return
 		}
 
+		if qs := r.URL.Query(); qs.Get("userId") != "" {
+			if isCallerAdmin == "false" {
```

Comment:
With a boolean and proper error checking this condition will be much safer, but if you keep it as is at least invert this to `isCallerAdmin != "true"` so we don't fallthrough assuming it is an admin on anything other than a literal `"false"` (incl empty)